src/benchmark/benchmark.ts

```diff
@@ -65,12 +68,10 @@ class Benchmark {
         Benchmark.executeJS(func[0], args, expectedResult, warmupRounds);
         const jsTimes = Benchmark.executeJS(func[0], args, expectedResult, measureRounds);
 
-        const wasmArgs = args.slice();
-        wasmArgs.unshift(func[0].name);
-
-        const wasmAlgorithm = this.transpiler.transpile(func.join(''));
-        this.executeWasm(wasmAlgorithm, wasmArgs, expectedResult, warmupRounds);
-        const wasmTimes = this.executeWasm(wasmAlgorithm, wasmArgs, expectedResult, measureRounds);
+        const callWrapper = transpiler.transpile(func.join(''));
+        const wasmAlgorithm = callWrapper.setCallFunctionName(func[0].name).call.bind(callWrapper);
```

Comment:
Why is this `bind` call here necessary?

Reply:
Because we extract the call function and run it on it's own. The this context is therefore undefined. I changed it so that no longer the function but the callwrapper is passed to the execute method.